src/ar_firewall.erl

```diff
@@ -77,6 +150,60 @@ load_blacklist_test() ->
 					binary = <<"BADCONTENT">>
 				}
 		},
+	{Sigs, _, _} = av_sigs:load(["test/test_sig.ndb"]),
+	?assertEqual([ExpectedSig], Sigs),
 	ar_meta_db:put(content_policies, ["test/test_sig.ndb"]),
-	Sigs = av_sigs:all(),
-	?assertEqual([ExpectedSig], Sigs).
+	ar_firewall:reload(),
+	?assertEqual(reject, scan_tx((ar_tx:new())#tx{ data = <<".. BADCONTENT ..">> })),
+	?assertEqual(accept, scan_tx((ar_tx:new())#tx{ data = <<".. B A D C ONTENT ..">> })).
+
+parse_txt_blacklist_test() ->
+	ExpectedSigs = [
+		#sig {
+			type = binary,
+			data =
+				#binary_sig {
+					offset = any,
+					binary = <<"BADCONTENT1">>
+				}
+		},
+		#sig {
+			type = binary,
+			data =
+				#binary_sig {
+					offset = any,
+					binary = <<"BADCONTENT2">>
+				}
+		},
+		#sig {
+			type = binary,
+			data =
+				#binary_sig {
+					offset = any,
+					binary = <<"BADCONTENT3">>
+				}
+		}
+	],
+	{Sigs, _, _ } = av_sigs:load(["test/test_sig.txt"]),
+	lists:foreach(
+		fun(ExpectedSig) ->
+			?assert(
+				lists:member(ExpectedSig, Sigs),
+				io:format("Expected ~p to be in ~p", [ExpectedSig, Sigs])
```

Comment:
Nice! We should do this more 👍👍

Reply:
Wait a sec, you probably wanted to use `io_lib:format/2` and not `io:format/2` :) Obviously this is not tested, so I wonder if it really is alright to have an iolist as a comment. I think we should render it into a string.

Reply:
No, `assert` does not print the result of `io_lib:format` in case of a failure but it does execute the code so `io:format` does the trick.

Reply:
✅